### ISSUE TITLE
Add NPY rule: prefer np.allclose(a, b) over np.all(np.isclose(a, b))

### ISSUE THREAD

**Issue by anilbey** [astral-sh/ruff#24948]
### Summary

It would be useful to have a rule under the `NPY` category that flags `np.all(np.isclose(...))` and suggests `np.allclose(...)` as a fix.

### Motivation

`np.allclose(a, b)` is a direct shorthand for `np.all(np.isclose(a, b))` — it's documented as such in the [NumPy reference](https://numpy.org/doc/stable/reference/generated/numpy.allclose.html), returns a Python `bool` instead of a 0-d array, and is more idiomatic. The two forms accept the same `rtol`/`atol`/`equal_nan` parameters, so the rewrite is mechanical and safe.

This pattern shows up frequently in test suites (e.g. `assert np.all(np.isclose(x, y))`) where authors reach for `isclose` first and wrap it in `all` rather than recalling that `allclose` exists.

### Example

```python
# Bad
assert np.all(np.isclose(values, expected))
assert np.all(np.isclose(values, expected, rtol=1e-5))

# Good
assert np.allclose(values, expected)
assert np.allclose(values, expected, rtol=1e-5)
```

**Comment by VascoSch92** [comment]
Hey, 

I scoped a little the rule and it is feasible and the rewrite is sound by construction. In fact, `np.allclose` is literally implemented as `np.all(np.isclose(...)):`

```python
def allclose(a, b, rtol=1.e-5, atol=1.e-8, equal_nan=False):
    res = all(isclose(a, b, rtol=rtol, atol=atol, equal_nan=equal_nan))
    return builtins.bool(res)
```

Remarks:

1. Only fire when `np.all` has a single argument. `np.all` accepts `axis/keepdims/where/out`; np.allclose doesn't.
2. the autofix should be marked as unsafe as the return type differs

All the machinery already exists in the NPY rules: alias-safe qualified-name resolution (NPY002) and autofixes (NPY003/NPY201), plus C419 as a precedent for the nested-call detect-and-rewrite pattern.

Do we still want to create this rule?
